tasks/request_files/request_files.py

```diff
@@ -235,88 +244,114 @@ def inner_task(event: Dict, max_retries: int, retry_sleep_secs: float,
+                'err_msg' (str): Will be modified if error occurs.
+
+
+        glacier_bucket: The S3 glacier bucket name.
+        restore_expire_days:
+            The number of days the restored file will be accessible in the S3 bucket before it expires.
+        max_retries: todo
+        retry_sleep_secs: todo
+        retrieval_type: todo
+        db_queue_url: todo
+        job_id: The unique identifier used for tracking requests.
+
+    Raises: RestoreRequestError if any file restore could not be initiated.
     """
-    request_time = datetime.now(timezone.utc).isoformat()
     attempt = 1
     granule_id = granule[GRANULE_GRANULE_ID_KEY]
 
-    post_status_for_job_to_queue(job_id, granule_id, ORCA_STATUS_PENDING, request_time, None, glacier_bucket,
-                                 RequestMethod.POST,
-                                 db_queue_url,
-                                 max_retries, retry_sleep_secs)
-
     while attempt <= max_retries + 1:
         for a_file in granule[GRANULE_RECOVER_FILES_KEY]:
             if not a_file[FILE_SUCCESS_KEY]:
                 try:
-                    restore_object(s3, a_file[FILE_KEY_KEY], restore_expire_days, glacier_bucket, attempt, job_id,
-                                   retrieval_type)
-                    a_file[FILE_SUCCESS_KEY] = True
-                    a_file[FILE_ERROR_MESSAGE_KEY] = ''
-
-                    post_status_for_file_to_queue(
-                        job_id, granule_id, os.path.basename(a_file[FILE_KEY_KEY]),
+                    restore_object(
+                        s3,
                         a_file[FILE_KEY_KEY],
-                        a_file[FILE_DEST_BUCKET_KEY],
-                        ORCA_STATUS_PENDING,
-                        None,
-                        request_time,
-                        datetime.now(timezone.utc).isoformat(),
+                        restore_expire_days,
+                        glacier_bucket,
+                        attempt,
+                        job_id,
+                        retrieval_type,
+                    )
+                    a_file[FILE_SUCCESS_KEY] = True
+                    a_file[FILE_ERROR_MESSAGE_KEY] = ""
+
+                    files = [
+                        {"filename": os.path.basename(a_file[FILE_KEY_KEY])},
+                        {"key_path": a_file[FILE_KEY_KEY]},
+                        {"restore_destination": a_file[FILE_DEST_BUCKET_KEY]},
+                        {"status_id": shared_recovery.OrcaStatus.PENDING.value},
+                        {"error_message": None},
+                        {"request_time": datetime.now(timezone.utc).isoformat()},
+                        {"last_update": datetime.now(timezone.utc).isoformat()},
+                        {"completion_time": datetime.now(timezone.utc).isoformat()},
+                    ]
+                    shared_recovery.create_status_for_job(
```

Comment:
Since this is expecting an array that contains all the files, I wonder if this should not be called earlier or possibly at the end after all the final status information is needed.

Reply:
Sounds good to me. There will be enough of a delay that there won't be a race condition.